Fix falsy health checks

tim-ebert opened the pull request:
**What this PR does / why we need it**:
Before, `grm`'s MR controller and health controller were acting on MRs independent of each other / in parallel.
This could lead to a situation, were the health controller checks the resource first, sees no resources in `.status.resources` and therefore sets the `ResourcesHealthy` condition to `True` immediately. Later the MR controller successfully applies the resources and sets the `ResourcesApplied` condition to `True`. Now the MR has both conditions set to `True`, which indicates, that is is completely reconciled and healthy, although resources may still be creating (e.g. Pods for a Deployment). Later the health controller will check the resource again and correct this false condition.

This means, for up to `--health-sync-period` the managed resource may indicate a false health status, which is problematic for controllers waiting for a MR to be "Ready" (for example gardenlet's `ControllerInstallation` controller).

With this PR, the MR controller invalidates the `ResourcesHealthy` condition (i.e. sets it to `Progressing`) and the health checks are skipped until the `ResourcesApplied` condition is set to `True`, which indicates that the resource is now fully reconciled.


**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
I fixed a bug along the way, which I introduced in #47, which caused ignored resources not to be created.

**Release note**:
<!--  Write your release note:
1. Enter your release note in the below block.
2. If no release note is required, just write "NONE" within the block.

Format of block header: <category> <target_group>
Possible values:
- category:       improvement|noteworthy|action
- target_group:   user|operator|developer
-->
```improvement operator
A bug has been fixed, which could lead to a situation, where a `ManagedResource` is falsely indicating a "Ready" state for a short period of time.
```

tim-ebert commented:
@gardener/gardener-maintainers has anyone time for a review?

rfranzke commented on pkg/controller/managedresources/controller.go:
```diff
@@ -236,49 +235,65 @@ func (r *Reconciler) reconcile(mr *resourcesv1alpha1.ManagedResource, log logr.L
+		reason := "ApplyProgressing"
+		msg := "The resources are currently being reconciled."
+		switch conditionResourcesApplied.Reason {
+		case "ApplyFailed", "DeletionFailed", "DeletionPending":
```
Now that you rely on the reasons, can you introduce constants for them?

tim-ebert commented:
Thanks for your review!
I have addressed your comments, PTAL :)

Also changed:
- renamed reason `HealthChecksProgressing` -> `HealthChecksPending` as they weren't really "Progressing" but rather skipped/pending in that case
- combined reason `DeletionProgressing` and `DeletionPending` to `DeletionPending`
